src/dynamics/ads_player/ads_player.js

```diff
@@ -58,6 +58,11 @@ Scoped.define("module:Ads.Dynamics.Player", [
                         } else {
                             return this.adsManager.setVolume(Maths.clamp(volume, 0, 1));
                         }
+                    },
+                    "change:endcardbackgroundsrc": function(endcardbackgroundsrc) {
```

Comment:
Attaching `change:endcardbackgroundsrc` for all cases is not could be efficient as will load with non-mandatory listener. You can listen it only if it's really required. Like when is true `this.shouldShowFirstFrameAsEndcard()` . Locally you can set like `this.on("change:endcardbackgroundsrc", function(state) => {/*your code here*/}, this);`

Reply:
:+1: added this logic to the create function